esbuild: Error when declaring relative path global scripts without starting them by "./" on Windows.

### Command

build, serve

### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

_No response_

### Description

This issue only happens on Windows, I couldn't reproduce this on Linux.

When importing global scripts with relative paths that don't start with "./" in angular.json , the commands "ng build" and "ng serve" fail with an error.

### Minimal Reproduction

- ng new test-project
- npm i marked
- Update angular.json to include:
"node_modules/marked/marked.min.js" making sure it's not "./node_modules/marked/marked.min.js"
- npm i

Changing it to 
"./node_modules/marked/marked.min.js" gets rid of the error.

### Exception or Error

```text
X [ERROR] Could not resolve "node_modules/marked/marked.min.js" [plugin angular-global-scripts]
```


### Your Environment

```text
_                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/


Angular CLI: 16.0.1
Node: 16.18.0
Package Manager: npm 9.6.4
OS: win32 x64

Angular: 16.0.1
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1600.1
@angular-devkit/build-angular   16.0.1
@angular-devkit/core            16.0.1
@angular-devkit/schematics      16.0.1
@schematics/angular             16.0.1
rxjs                            7.8.1
typescript                      5.0.4
```


### Anything else relevant?

Windows only issue.

--- a/packages/angular_devkit/build_angular/src/builders/browser-esbuild/global-scripts.ts
+++ b/packages/angular_devkit/build_angular/src/builders/browser-esbuild/global-scripts.ts
@@ -10,6 +10,8 @@ import type { BuildOptions } from 'esbuild';
 import MagicString, { Bundle } from 'magic-string';
 import assert from 'node:assert';
 import { readFile } from 'node:fs/promises';
+import path from 'node:path';
+import { assertIsError } from '../../utils/error';
 import { NormalizedBrowserOptions } from './options';
 import { createSourcemapIngorelistPlugin } from './sourcemap-ignorelist-plugin';
 
@@ -96,23 +98,36 @@ export function createGlobalScriptsBundleOptions(
             // Global scripts are concatenated using magic-string instead of bundled via esbuild.
             const bundleContent = new Bundle();
             for (const filename of files) {
-              const resolveResult = await build.resolve(filename, {
-                kind: 'entry-point',
-                resolveDir: workspaceRoot,
-              });
+              let fileContent;
+              try {
+                // Attempt to read as a relative path from the workspace root
+                fileContent = await readFile(path.join(workspaceRoot, filename), 'utf-8');
+              } catch (e) {
+                assertIsError(e);
+                if (e.code !== 'ENOENT') {
+                  throw e;
+                }
 
-              if (resolveResult.errors.length) {
-                // Remove resolution failure notes about marking as external since it doesn't apply
-                // to global scripts.
-                resolveResult.errors.forEach((error) => (error.notes = []));
+                // If not found attempt to resolve as a module specifier
+                const resolveResult = await build.resolve(filename, {
+                  kind: 'entry-point',
+                  resolveDir: workspaceRoot,
+                });
 
-                return {
-                  errors: resolveResult.errors,
-                  warnings: resolveResult.warnings,
-                };
+                if (resolveResult.errors.length) {
+                  // Remove resolution failure notes about marking as external since it doesn't apply
+                  // to global scripts.
+                  resolveResult.errors.forEach((error) => (error.notes = []));
+
+                  return {
+                    errors: resolveResult.errors,
+                    warnings: resolveResult.warnings,
+                  };
+                }
+
+                fileContent = await readFile(resolveResult.path, 'utf-8');
               }
 
-              const fileContent = await readFile(resolveResult.path, 'utf-8');
               bundleContent.addSource(new MagicString(fileContent, { filename }));
             }
 

